server/store/memory/ip_store.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package memory
+
+import (
+	"net"
+	"sync"
+
+	"github.com/chihaya/chihaya/server/store"
+)
+
+func init() {
+	store.RegisterIPStoreDriver("memory", &ipStoreDriver{})
+}
+
+type ipStoreDriver struct{}
+
+func (d *ipStoreDriver) New(cfg *store.Config) (store.IPStore, error) {
+	return &ipStore{
+		ips: make(map[[16]byte]struct{}),
+	}, nil
+}
+
+// ipStore implements store.IPStore using an in-memory map of byte arrays
+type ipStore struct {
+	ips map[[16]byte]struct{}
+	sync.RWMutex
+}
+
+var (
+	_            store.IPStore = &ipStore{}
+	v4InV6Prefix               = []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xff, 0xff}
+)
+
+// toBytes converts an IP address to a [16]byte.
+// If an IPv4 address is specified, it will be prefixed with
+// the net.v4InV6Prefix and thus becomes a valid IPv6 address.
+func toBytes(ip net.IP) [16]byte {
```

Comment:
Add something in the comment that they need to be an array so that it can be a key for the map.